File: src/test/java/leets/leenk/domain/feed/domain/service/CommentDeleteServiceTest.java
package leets.leenk.domain.feed.domain.service;

import leets.leenk.domain.feed.domain.entity.Comment;
import leets.leenk.domain.feed.domain.entity.Feed;
import leets.leenk.domain.feed.test.CommentTestFixture;
import leets.leenk.domain.feed.test.FeedTestFixture;
import leets.leenk.domain.feed.test.UserTestFixture;
import leets.leenk.domain.user.domain.entity.User;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.time.LocalDateTime;

import static org.assertj.core.api.Assertions.assertThat;

public class CommentDeleteServiceTest {
    private final CommentDeleteService commentDeleteService = new CommentDeleteService();

    @Test
    @DisplayName("CommentDeleteService 테스트")
    void CommentDeleteService() {
        // given
        User user = UserTestFixture.createUser(1L, "me");
        Feed feed = FeedTestFixture.createFeed(1L, user);
        Comment comment = CommentTestFixture.createComment(null, user, feed, "hi");

        // when
        LocalDateTime start = LocalDateTime.now();

        commentDeleteService.deleteComment(comment);

        LocalDateTime end = LocalDateTime.now();

        // then
        assertThat(comment.getDeletedAt()).isNotNull();
        assertThat(comment.getDeletedAt()).isAfterOrEqualTo(start);
        assertThat(comment.getDeletedAt()).isBeforeOrEqualTo(end);
    }
}

File: src/test/kotlin/leets/leenk/domain/feed/domain/service/CommentDeleteServiceTest.kt
package leets.leenk.domain.feed.domain.service

import io.kotest.core.spec.style.BehaviorSpec
import io.kotest.matchers.shouldBe
import io.kotest.matchers.shouldNotBe
import leets.leenk.domain.feed.test.CommentTestFixture
import leets.leenk.domain.feed.test.FeedTestFixture
import leets.leenk.domain.feed.test.UserTestFixture
import java.time.LocalDateTime

class CommentDeleteServiceTest :
    BehaviorSpec({
        val commentDeleteService = CommentDeleteService()

        Given("사용자, 피드, 댓글이 주어졌을 때") {
            val user = UserTestFixture.createUser(1L, "me")
            val feed = FeedTestFixture.createFeed(1L, user)
            val comment = CommentTestFixture.createComment(1L, user, feed, "hi")

            When("댓글을 삭제하면") {
                val start = LocalDateTime.now()
                commentDeleteService.deleteComment(comment)
                val end = LocalDateTime.now()

                Then("deletedAt이 현재 시간으로 설정되어야 한다") {
                    comment.deletedAt shouldNotBe null
                    (comment.deletedAt!! >= start) shouldBe true
                    (comment.deletedAt!! <= end) shouldBe true
                }
            }
        }
    })
